frontend/src/components/PopUpLeagues.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { Box, Button, styled, TextField } from '@mui/material';
+import { leagueDataSource } from '../datasources/index';
+
+const PopUpContainer = styled(Box)({
+    height: "fit-content",
+    padding: "15px",
+    display: "flex", 
+    alignItems: "center",
+    flexDirection: "column",
+    gap: "10px"
+});
+
+const CustomButton = styled(Button)({
+    backgroundColor: " #E5B226",
+    color: "#FBF9F9",
+    boxShadow: "0px 3px #D9D9D9",
+    borderRadius: "10px",
+    fontWeight: "bold",
+    ":hover":{
+        backgroundColor: "rgb(140, 109, 23)",
+        boxShadow: "none"
+    }
+});
+
+interface PopupProps {
+    selectedButton: string
+    onClose : (value: boolean) => void
+    onUpdate : () => void
+};
+
+export function PopUpLeagues(props: PopupProps){
+
+    const [leagueName, setLeagueName] = useState<string>("");
+    const [leagueId, setLeagueId] = useState<string>("");
```

Comment:
etant donne que les deux sont jamais visibles en meme temps, 1 seul variable peut gerer les deux
```suggestion
     const [userInput, setUserInput] = useState<string>("");
```